### PR TITLE
Pass metadata for factor creation methods

### PR DESCRIPTION
**Pull request by IldarAbdullin-okta**
### Problem Analysis (Technical)
Metadata is required for signed_nonce and totp factors

### CODE REVIEW

File: Sources/DeviceAuthenticator/Networking/LegacyServerAPI.swift [hunk 1]
```diff
@@ -229,16 +233,20 @@ class LegacyServerAPI: ServerAPIProtocol {
...
+                                                             metadata: metadata,
+                                                             and: method) else {
             return nil
         }
 
         return pushFactor
     }
 
     func createEnrolledPushFactor(from factorModels: [EnrollingFactor],
+                                  metadata: AuthenticatorMetaDataModel,
```
Comment (moisesolmedo-okta) [inline Sources/DeviceAuthenticator/Networking/LegacyServerAPI.swift:249]: are you using it somewhere?

Comment (IldarAbdullin-okta) [reply Sources/DeviceAuthenticator/Networking/LegacyServerAPI.swift:249]: pls read description. totp needs it

Comment (moisesolmedo-okta) [reply Sources/DeviceAuthenticator/Networking/LegacyServerAPI.swift:249]: yes but don't see it being used in `createEnrolledPushFactor `

**Review by moisesolmedo-okta — APPROVED** [review]

File: Sources/DeviceAuthenticator/Networking/LegacyServerAPI.swift (hunk 1 above)
Comment (IldarAbdullin-okta) [reply Sources/DeviceAuthenticator/Networking/LegacyServerAPI.swift:249]: Push factor doesn't need it so far, but TOTP in internal sdk still needs it. I really don't want to save metadata as a property so we can still keep API instance without states. Override enrollment and update requests in internal SDK also doesn't look as a good idea

**Pull request merged**